test/automation/search_automation_test.go

```diff
@@ -0,0 +1,128 @@
+package automation
+
+import (
+	"fmt"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/stretchr/testify/suite"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+type SearchAutomationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *SearchAutomationTestSuite) TestSearch_NoArg() {
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.Spawn("search")
+	cp.ExpectLongString("The following argument is required:")
+	cp.ExpectExitCode(1)
+
+	cp = ts.Spawn("search", "--language", "python")
+	cp.ExpectLongString("The following argument is required:")
+	cp.ExpectExitCode(1)
+
+}
+
+func (suite *SearchAutomationTestSuite) TestSearch_NoLanguageArg() {
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.Spawn("search", "--language")
+	cp.ExpectLongString("flag needs an argument: --language")
+	cp.ExpectExitCode(1)
+
+}
+
+func (suite *SearchAutomationTestSuite) TestSearch_OutProject() {
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cp := ts.Spawn("search", "flask")
+	cp.ExpectLongString("Language must be provided by flag or by running this command within a project.")
+	cp.ExpectExitCode(1)
+
+}
+
+func (suite *SearchAutomationTestSuite) TestSearch_Flask() {
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	url := "https://platform.activestate.com/qamainorg/public?branch=main&commitID=32e543ee-b6ab-4f59-9b28-ad830ec6980e"
+	suite.Require().NoError(fileutils.WriteFile(filepath.Join(ts.Dirs.Work, "activestate.yaml"), []byte("project: "+url)))
```

Comment:
Was this necessary? IIRC you can use `state search` without a project.

Reply:
Using search without `--language` should generate an error in some cases including this one.
Another use is verification which in all situations it provides required info.